### DISCUSSION TITLE
Host instances externally on different ports?

### DISCUSSION THREAD

**Discussion by realJogicodes** [pocketbase/pocketbase#4095] [Q&A] [answered]
Hi! I love this project, it is a superior backend to supabase in scope and execution for me personally because pocketbase is truly open source and portable, whereas supabase can be moved in theory if they rug pull me, but not so much in practice. 

Anyway, here's my question, first as TLDR, then a little explain as to why:

### TLDR for my question:

Could I run 3x pocketbase on a linode server fetching from

-  `https://[linode-server-ip-address]:1234`
-  `https://[linode-server-ip-address]:1235`
-  `https://[linode-server-ip-address]:1236`

Where I have 3 pocketbase instances running on the same server for 3 different very low traffic projects? How?

### Why would I want that?

Doing so would be a modular approach that is cost effective (minimal servers are $5/month) while I throw spaghetti against the wall and once something finally sticks and takes off (fingers crossed) I can move that pocketbase instance into a dedicated server if need be or just crank up the server specs

Because structurally, I would like to host my front ends and my back ends separately. My front end use svelte kit on edge functions on say vercel, netlify, github pages, cloudflare, heroku (there are so many options) and with the pocketbase backend on a dedicated linode server with persistent IP, exposed via custom port so that more than one pocketbase instance can run. 

Can that be done? I went through the docs to no avail and google gives me this, an old discussion referencing something like this, but the link in it is dead:
https://github.com/pocketbase/pocketbase/discussions/321

**Comment by ganigeorgiev** [comment, marked as answer]
You can start multiple PocketBase processes with different `pb_data` dir and port by specifying the `--dir` and `--http` flag:

```sh
# for all available flags check ./pocketbase serve --help
./pocketbase serve --dir="./pb_data1" --http="127.0.0.1:8091" & \
./pocketbase serve --dir="./pb_data2" --http="127.0.0.1:8092" &
...

# to kill all instances:
sudo killall pocketbase
```

But note that the above will not work with the autoissued Let's encrypt because the verification must happen on 80/443 and you can't have multiple applications binding to the same port (_unless `SO_REUSE_ADDRESS` or other os socket option is used_).
If you want to host multiple applications on the same server, my recommendation is to have a reverse proxy (nginx, apache, caddy, etc.) that will be the only one binding to 80/443 and manage the certificate with [certbot](https://certbot.eff.org/) or similar. For an example nginx config you could check https://pocketbase.io/docs/going-to-production/#using-reverse-proxy.
